django/bossspatialdb/views.py

```diff
@@ -488,3 +488,117 @@ def delete(self, request, collection, experiment, channel):
+        """
+        View to handle PUT requests for a overwriting a cuboid to 0s 
+
+        Due to parser implementation, request.data should be a numpy array already.
+
+        :param request: DRF Request object
+        :type request: rest_framework.request.Request
+        :param collection: Unique Collection identifier, indicating which collection you want to access
+        :param experiment: Experiment identifier, indicating which experiment you want to access
+        :param channel: Channel identifier, indicating which dataset or annotation project you want to access
+        :param resolution: Integer indicating the level in the resolution hierarchy (0 = native)
+        :param x_range: Python style range indicating the X coordinates of where to post the cuboid (eg. 100:200)
+        :param y_range: Python style range indicating the Y coordinates of where to post the cuboid (eg. 100:200)
+        :param z_range: Python style range indicating the Z coordinates of where to post the cuboid (eg. 100:200)
+        :return:
+        """
+
+        # Check for optional iso flag
+        if "iso" in request.query_params:
+            if request.query_params["iso"].lower() == "true":
+                iso = True
+            else:
+                iso = False
+        else:
+            iso = False
+
+        try:
+            request_args = {
+                "service": "cutout",
+                "collection_name": collection,
+                "experiment_name": experiment,
+                "channel_name": channel,
+                "resolution": resolution,
+                "x_args": x_range,
+                "y_args": y_range,
+                "z_args": z_range,
+                "time_args": t_range,
+            }
+            req = BossRequest(request, request_args)
+        except BossError as err:
+            return err.to_http()
+
+        # Convert to Resource
+        resource = project.BossResourceDjango(req)   
+        
+        # Get data type
+        try:
+            self.bit_depth = resource.get_bit_depth()
+            expected_data_type = resource.get_numpy_data_type()
+        except ValueError:
+            return BossHTTPError("Unsupported data type: {}".format(resource.get_data_type()), ErrorCodes.TYPE_ERROR)
+    
+        # Set a limit of CUTOUT_MAX_SIZE 
+        if is_too_large(req, self.bit_depth):
+            return BossHTTPError("Cutout overwrite is over 500MB when uncompressed. Reduce overwrite dimensions.", ErrorCodes.REQUEST_TOO_LARGE)
+    
+        # Get the shape of the requested data clear
+        if len(req.get_time()) > 2:
+            expected_shape = (len(req.get_time()), req.get_z_span(), req.get_y_span(), req.get_x_span())
+        else:
+            expected_shape = (req.get_z_span(), req.get_y_span(), req.get_x_span())
+
+        # Create a binary numpy array for overwrite with specified shape and dtype
+        black_cuboid = np.ones(expected_shape, dtype=expected_data_type)
```

Comment:
You say in the class comment that you are overriding with zeros, but you use `np.one` which I assume creates an array of ones...

Reply:
Yep so whats really going on is that I'm using an array of ones to denote what voxels I have to overwrite with a 0. This black_cuboid is sent to the data buffer in SPDB which is initialized with `np.zeros` so in order to differentiate the voxels the user wants to delete from the other voxels in the cuboid I use ones. I will make that more clearer in the doc strings.

Reply:
It might be helpful to have a comment describing this, as it is not fully intuitive (until you grok the all of the code).